tests/unit/test_schema_legacy.py

```diff
@@ -0,0 +1,195 @@
+    "valid_name",
+    [
+        "this",
+        "that",
+        "walker_turzai",
+        "foo_bar_baz",
+        "name123_four",
+        "r1",
+        "crush_",
+    ],
+)
+def test_valid_role_name(galaxy_info, valid_name):
+    galaxy_info["role_name"] = valid_name
+    info = LegacyGalaxyInfo(**galaxy_info)
+    assert info.role_name == valid_name
+
+
+@pytest.mark.parametrize(
+    "invalid_name",
+    [
+        "_this",
+        "walker-turzai",
+        "foo_bar-baz",
+        "3w6",
+        "$@#",
+        "this.role",
+        "docker!",
+        "big space",
+        "whyUpper",
+    ],
+)
+def test_invalid_role_name(galaxy_info, invalid_name):
+    galaxy_info["role_name"] = invalid_name
+    with pytest.raises(ValueError, match=re.escape(f"role name {invalid_name} is invalid")):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_author(galaxy_info):
+    galaxy_info["author"] = galaxy_info["author"] * 100
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_description(galaxy_info):
+    galaxy_info["description"] = "Description!" * 100
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_company(galaxy_info):
+    galaxy_info["company"] = "Red Hat" * 10
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_url(galaxy_info):
+    galaxy_info["issue_tracker_url"] = galaxy_info["issue_tracker_url"] + ("/a" * 1001)
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_license(galaxy_info):
+    galaxy_info["license"] = galaxy_info["license"] * 30
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_version(galaxy_info):
+    galaxy_info["min_ansible_version"] = galaxy_info["min_ansible_version"] * 50
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+    galaxy_info["min_ansible_version"] = "2.1"
+    galaxy_info["min_ansible_container_version"] = galaxy_info["min_ansible_container_version"] * 50
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+def test_max_tag(galaxy_info):
+    galaxy_info["galaxy_tags"][0] = "rhel" * 100
+    with pytest.raises(ValueError, match="must not exceed"):
+        LegacyGalaxyInfo(**galaxy_info)
+
+
+@pytest.mark.parametrize(
+    "valid_dependency",
+    [
+        [],
+        ["geerlingguy.php"],
+        ["eamontracey.hello_role"],
+        ["geerlingguy.php", "eamontracey.hello_role"],
+    ],
+)
+def test_valid_dependencies(galaxy_info, valid_dependency):
+    metadata = LegacyMetadata(LegacyGalaxyInfo(**galaxy_info), valid_dependency)
+    assert metadata.dependencies == valid_dependency
+
+
+def test_invalid_dependency_type(galaxy_info):
```

Comment:
Are there any other incorrect types you could add here under a `@pytest.mark.parametrize`? If you add string type like (where meta/main.yml has `dependencies: geerlingguy.java`) it doesn't error about type, but instead `ValueError: g must have namespace and name separated by '.'`